domain/consensus/utils/pow/pow.go

```diff
@@ -12,52 +12,76 @@ import (
 	"math/big"
 )
 
-// CheckProofOfWorkWithTarget check's if the block has a valid PoW according to the provided target
-// it does not check if the difficulty itself is valid or less than the maximum for the appropriate network
-func CheckProofOfWorkWithTarget(header externalapi.MutableBlockHeader, target *big.Int) bool {
-	// The block pow must be less than the claimed target
-	powNum := CalculateProofOfWorkValue(header)
-
-	// The block hash must be less or equal than the claimed target.
-	return powNum.Cmp(target) <= 0
+type MinerState struct {
```

Comment:
Can you rename it to something more mining agnostic? Maybe just `State`? (and then it'll be `pow.State`).
It's used as part of validation, so the mining terminology feels unrelated.